lightningd/log.c

```diff
@@ -291,13 +291,15 @@ struct log_book *new_log_book(struct lightningd *ld, size_t max_mem)
 }
 
 static enum log_level filter_level(struct log_book *lr,
-				   const struct log_prefix *lp)
+				   const struct log_prefix *lp,
+				   const struct node_id *node_id)
 {
 	struct print_filter *i;
+	const char *node_id_str = node_id ? node_id_to_hexstr(tmpctx, node_id) : "";
```

Comment:
I wonder if doing it the other way around, i.e., converting the prefix to binary and matching at byte-level, wouldn't be more efficient, rather than converting the node_id to hex over and over again. That's with a caveat though that the prefix would need to have an even length.

Alternatively we could also keep a list of prior matches.

Reply:
We can generate this on-stack pretty fast, but I think optimizing logging CPU to this level is probably overkill for now?